src/oracle/connection/raw.rs

```diff
@@ -1,45 +1,62 @@
-use std::str;
 use oci_sys as ffi;
+use std::ffi::{CStr, CString};
 use std::os::raw as libc;
 use std::ptr;
+use std::str;
 
 use diesel::result::*;
 
 pub struct ConnectionEnviroment {
     handle: *mut ffi::OCIEnv,
     pub error_handle: *mut ffi::OCIError,
+    pub cs_id: u16,
 }
 
 impl ConnectionEnviroment {
     pub fn new() -> Result<ConnectionEnviroment, ()> {
         let env_handle = unsafe {
             let mut handle: *mut ffi::OCIEnv = ptr::null_mut();
-            ffi::OCIEnvNlsCreate(&mut handle as *mut _,
-                                 ffi::OCI_DEFAULT,
-                                 ptr::null_mut(),
-                                 None,
-                                 None,
-                                 None,
-                                 0,
-                                 ptr::null_mut(),
-                                 0,
-                                 0);
+            let code = ffi::OCIEnvNlsCreate(
+                &mut handle as *mut _,
+                ffi::OCI_DEFAULT,
+                ptr::null_mut(),
+                None,
+                None,
+                None,
+                0,
+                ptr::null_mut(),
+                0,
+                0,
+            );
+            if code != 0 {
+                println!("Couldn't create environment");
```

Comment:
I think we should err out here, right? Nothing you can do without the environment handle.